packages/flutter/lib/src/rendering/table.dart

```diff
@@ -595,7 +595,7 @@ class RenderTable extends RenderBox {
   /// children are simply moved to their new location in the table rather than
   /// removed from the table and re-added.
   void setFlatChildren(int columns, List<RenderBox?> cells) {
-    if (cells == _children && columns == _columns)
+    if (listEquals(cells, _children) && columns == _columns)
```

Comment:
This is an optimization of duplicate check logic

Reply:
Have you verified that this is actually faster? It trades a pointer comparison for an O(N) comparison during updates to avoid an O(N) cost in layout, but it's not clear to me how common it would be for the child list to be a new instance but holding the same contents.

Reply:
Currently, the caller of `setFlatChildren`always generate a new List instant so this is a dead code never reachable.

Next, let's analyze different scenarios:
1. Update `TableRow` widgets without changing its Type, so the `cells` parameter will have the same renderObjects add Will achieve optimized results.
2. Change the `TableRow` widgets count will have no effect because `listEquals` will compares length first.
3. Update ``TableRow` widgets with a different renderObjectWidget and keep the same count，like you said will increase the time complexity.

In summary, I believe that the benefits of such a modification are greater than the costs, Of course, also need to listen to your suggestions ：）

Reply:
If we always call with a new list, then this is converting an O(1) equality test with an O(N) equality test, which is slower, in order to avoid the code below, which is O(N) (but when avoided becomes O(1), so faster). This seems like it should be better... but my question is, is it?

The way to find out would be to benchmark it.